Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/Domain/Action/UI/Location/UpdateLocation.hs

```diff
@@ -150,9 +151,10 @@ updateLocationHandler UpdateLocationHandle {..} waypoints = withLogTag "driverLo
             mbRideIdAndStatus
     pure Success
   where
-    filterNewWaypoints mbOldLoc = do
+    filterNewWaypointsList mbOldLoc = do
+      currTime <- getCurrentTime
       let sortedWaypoint = toList $ NE.sortWith (.ts) waypoints
-      maybe sortedWaypoint (\oldLoc -> filter ((oldLoc.coordinatesCalculatedAt <) . (.ts)) sortedWaypoint) mbOldLoc
+      pure $ maybe sortedWaypoint (\oldLoc -> filter (\newPoint -> oldLoc.coordinatesCalculatedAt < newPoint.ts && newPoint.ts <= currTime) sortedWaypoint) mbOldLoc
```

Comment:
can you add a buffer time here, i.e. `newPoint.ts <= currTime + bufferMinutes (config)`
you can use the same config as used in driverPool for checking last active location update i.e. `driverPoolCfg.driverPositionInfoExpiry`